pkg/config/constants.go

```diff
@@ -42,6 +42,9 @@ const (
 	// WorkspaceIDLabel is label key to store workspace identifier
 	WorkspaceIDLabel = "che.workspace_id"
 
+	// WorkspaceEndpointNameLabel is the label key for storing an endpoint's name from the devfile representation
+	WorkspaceEndpointNameLabel = "che.workspace.endpoint_name"
```

Comment:
It would mean that we're making endpoint name DNS name compatible. Do we really want to do it?
AFAIS we don't fetch objects by this label, so we're able to use annotation which is less strict in the values format.

Storing WorkspaceEndpointName would mean that we create one Ingress/Route per endpoint but after we reached Route number limit on OSIO, we started reuse Routes if possible...
Maybe we should store EndpointNames instead.

Reply:
Makes sense, though note that we're already making endpoint names DNS compatible, since we use them as a part of the route/ingress name.

We can assign them as annotations as well, I'll do that.

I'm not clear on what you mean by 
> Storing WorkspaceEndpointName would mean that we create one Ingress/Route per endpoint but after we reached Route number limit on OSIO, we started reuse Routes if possible...

On OpenShift, routes only support a single port, so we'll need to create one per endpoint in any case. I would consider merging endpoints into a single route (which I suppose we could do with e.g. the webviews endpoint) as a separate PR -- WDYT?

Reply:
Well, theia and webview it's another case since we need unique hosts there and it's why the corresponding attribute is there https://github.com/eclipse/che-plugin-registry/blob/master/v3/plugins/eclipse/che-theia/7.11.0/meta.yaml#L33

I briefly checked our plugins and devfile and sharing host by endpoint is more imaginary case then currently used.
But theoretically I wonder if it makes sense for cloud-shell to have two different endpoint:
```
spec:
  endpoints:
   -  name: cloud-shell
      public: true
      targetPort: 4444
      attributes:
        protocol: http
        type: ide
        discoverable: false
        secure: true
        cookiesAuthEnabled: true
   -  name: machine-exec
      public: true
      targetPort: 4444
      attributes:
        protocol: ws
        type: terminal
        discoverable: false
        secure: true
        cookiesAuthEnabled: true
```

>  I would consider merging endpoints into a single route (which I suppose we could do with e.g. the webviews endpoint) as a separate PR -- WDYT?

:+1: 
If now we create Ingress/Route per endpoint - is then https://github.com/eclipse/che/issues/15990 actual?

Reply:
> If now we create Ingress/Route per endpoint - is then eclipse/che#15990 actual?

Technically no, but I can't test since moving past 7.8.0 is blocked by https://github.com/eclipse/che/issues/16568.